### ISSUE TITLE
Tasks fail when Gradle Daemon is activated

### ISSUE THREAD

**Issue by maffelbaffel** [marklogic-community/ml-gradle#337]
Some gradle tasks dont work as soon as a gradle daemon is used. In recent Gradle version the daemon is activated by default.

gradlew.bat mlDeploy --stacktrace
````
Caused by: java.lang.RuntimeException: Unable to copy file to string from path: C:\Dev\Projekte\ml-demo\src\main\ml-config\security\roles\1-ml-demo-nobody-role.json; cause: src\main\ml-config\security\roles\1-ml-demo-nobody-role.json
        at com.marklogic.appdeployer.command.AbstractCommand.copyFileToString(AbstractCommand.java:110)
        at com.marklogic.appdeployer.command.AbstractCommand.copyFileToString(AbstractCommand.java:124)
        at com.marklogic.appdeployer.command.AbstractCommand.saveResource(AbstractCommand.java:137)
        at com.marklogic.appdeployer.command.AbstractResourceCommand.processExecuteOnResourceDir(AbstractResourceCommand.java:79)
        at com.marklogic.appdeployer.command.AbstractResourceCommand.execute(AbstractResourceCommand.java:34)
        at com.marklogic.appdeployer.command.security.DeployRolesCommand.execute(DeployRolesCommand.java:43)
        at com.marklogic.appdeployer.impl.AbstractAppDeployer.executeCommand(AbstractAppDeployer.java:111)
        at com.marklogic.appdeployer.impl.AbstractAppDeployer.deploy(AbstractAppDeployer.java:70)
        at com.marklogic.appdeployer.AppDeployer$deploy.call(Unknown Source)
        at com.marklogic.gradle.task.DeployAppTask.deployApp(DeployAppTask.groovy:32)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        ... 30 more

Caused by: java.nio.file.NoSuchFileException: src\main\ml-config\security\roles\1-ml-demo-nobody-role.json
        at org.springframework.util.FileCopyUtils.copyToByteArray(FileCopyUtils.java:86)
        at com.marklogic.appdeployer.command.AbstractCommand.copyFileToString(AbstractCommand.java:107)
        ... 52 more
````

`gradlew.bat mlDeploy --stacktrace --no-daemon` **works** as indented.
This might be an actual Windows issue.

Env:
Windows 10
Gradle 4.6
ml-gradle: 3.6.1

**Comment by rjrudin** [comment]
As a test, try the following: run "gradle --stop" to stop all the daemons, and then try "gradle mlDeploy" from your project directory, allowing Gradle to fire up a daemon. 

My hunch is that this will work, as I'm guessing there's a Gradle daemon that was launched in a separate directory, and the file path is relative to that directory. 

I recall fixing a bug related to this before that involved loading modules, not resource files. It may be that ml-app-deployer needs a similar change to account for when the Gradle daemon is running in a separate directory. Ideally to me of course, the daemon would resolve files from where the directory that Gradle was launched in.

**Comment by maffelbaffel** [comment]
> As a test, try the following: run "gradle --stop" to stop all the daemons, and then try "gradle mlDeploy" from your project directory, allowing Gradle to fire up a daemon.

This does not fix the problem:
````
C:\Dev\Projekte\ml-demo>gradlew.bat --stop                   
Stopping Daemon(s)
1 Daemon stopped
C:\Dev\Projekte\ml-demo>gradlew.bat mlUndeploy -Pconfirm=true
Starting a Gradle Daemon, 1 stopped Daemon could not be reused, use --status for details

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':mlUndeployApp'.
> Unable to copy file to string from path: C:\Dev\Projekte\ml-demo\src\main\ml-config\rest-api.json; cause: src\main\ml-config\rest-api.json

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
1 actionable task: 1 executed
````

> It may be that ml-app-deployer needs a similar change to account for when the Gradle daemon is running in a separate directory. Ideally to me of course, the daemon would resolve files from where the directory that Gradle was launched in.

I think this is the only reasonable fix. I may be looking into this bug in more detail later, if i can.

**Comment by rjrudin** [comment]
Fingers crossed this is fixed via ml-app-deployer 3.6.3

**Comment by rjrudin** [comment]
@maffelbaffel ml-gradle 3.6.3 is now available if you'd like to try out the fix you made in ml-app-deployer
